tp1/tp1.go

```diff
@@ -24,17 +40,55 @@ type Carrito struct {
 // para cada super mercado, cuanto saldria comprar esos productos ahi.
 // Retorna un slice de carritos, donde se tiene uno para cada super mercado.
 func (p Productos) CalcularPrecios(ids ...int) []Carrito {
-	return nil
+	precioXSuper := map[string]int{}
+	prods := make(map[string]bool)
+	for _, id := range ids {
+		prods[strconv.Itoa(id)] = true
+	}
+	for _, prod := range p {
+		if _, present := prods[prod[1]]; present {
+			if precio, err := strconv.Atoi(prod[2]); err == nil {
+				precioXSuper[prod[0]] += precio
+			}
+		}
+	}
+	var res []Carrito
+	for k, v := range precioXSuper {
+		res = append(res, Carrito{k, v})
+	}
+	return res
 }
 
 // Promedio recibe el id de un producto y retorna el precio promedio
 // de ese producto usando los precios de todos los supermercados.
 func (p Productos) Promedio(idProducto int) float64 {
-	return 0
+	suma := 0
+	cantSuper := 0
+	for _, prod := range p {
+		if id, err := strconv.Atoi(prod[1]); err == nil && id == idProducto {
```

Comment:
No me parece que este mal, pero en go por lo general se intenta manejar el sad path first. Osea, handlear el caso en el que lo que estoy queriendo hacer falla. Por eso se suelen ver los tipicos `if err != nil`. Esto permite mejorar la indentacion del codigo, por ejemplo:
```go
err := operacion()
if err == nil {
    err := otraOperacion()
    if err == nil {
        // mas codigo
    }
}
```
Eso tiene muchas indentaciones, sin embargo se puede hacer asi:
```go
err := operacion()
if err != nil {
    return
}
err := otraOperacion()
if err != nil {
    return
}
// codigo
```
En esa ultima manera se deja el codgio real para el final de la funcion y se busca la condicion de corte mas cercana. Algunos le dicen "line of sight": https://www.youtube.com/watch?v=yeetIgNeIkc&t=256s

Reply:
Si, claramente ese código así escrito queda horrible. Lo malo de la alternative es que el código te queda muy verbose pero claramente entre las dos opciones la ultima es mejor porque lo hace mas fácil de mantener.